### DISCUSSION TITLE
How to integrate with Jenkins pipeline

### DISCUSSION THREAD

**Discussion by fezza74** [WasiqB/multiple-cucumber-html-reporter#389] [Q&A] [answered]
Hi all,
I made a new small Jenkins Pipeline project. Everything works fine, but I don't know how to integrate multiple-cucumber-html-reporter into jenkinsfile.

This is the file

```
pipeline {
    agent any
    
    tools { nodejs "node20" }
    
    stages {
        stage('Checkout') {
            steps {
                git branch: "test", url: 'https://github.com/fezza74/aisacsisco'
            }
        }
        stage('Build') {
            steps {
                sh 'npm i'
            }
        }
        stage('Pre Test'){
            steps {
                sh 'npm run pretest'
            }
        }
        stage('Test'){
            steps {
                sh 'npm run test'
            }
        }
    }
 
    post {
        failure {
                emailext(
                    subject: "${JOB_NAME}.${BUILD_NUMBER} FAILED",
                    mimeType: 'text/html',
                    to: "[email_address]",
                    body: readFile("test-results/report/index.html")
                )
            }
            success {
                emailext(
                    subject: "${JOB_NAME}.${BUILD_NUMBER} PASSED",
                    mimeType: 'text/html',
                    to: "[email_address]",
                    body: readFile("test-results/report/index.html")
                )
            }
    }
}
```
And this is the project structures
![Screenshot 2024-10-11 at 11 43 53](https://github.com/user-attachments/assets/7bdc3e33-7aff-44c4-b1a6-60a489a9e75d)

Where and how can I integrate the report?

Thank you very much

**Comment by WasiqB** [comment, marked as answer]
Hi @fezza74 ,

You can check out the sample project for [Playwright](https://github.com/WasiqB/multiple-cucumber-html-reporter/tree/main/examples/playwright) and the [GitHub Action workflow](https://github.com/WasiqB/multiple-cucumber-html-reporter/blob/main/.github/workflows/test-playwright.yml) to know how to implement it in CI.

In the meantime, you can try the new version [4.0.0-beta.0](https://github.com/WasiqB/multiple-cucumber-html-reporter/releases/tag/v4.0.0-beta.0).
